File: php-api/src/Controllers/OrganisationApiKeyController.php
<?php

declare(strict_types=1);

namespace Enkl\Api\Controllers;

use Enkl\Api\Db\Database;
use Enkl\Api\Services\OrganisationApiKeyService;
use Psr\Http\Message\ResponseInterface as Response;
use Psr\Http\Message\ServerRequestInterface as Request;

/** Ported from Controllers/OrganisationApiKeyController.cs. Same OrgAdmin gating and
 * callerOrgId() idiom as OrganisationSsoConfigController. */
final class OrganisationApiKeyController extends BaseController
{
    private function service(): OrganisationApiKeyService
    {
        return new OrganisationApiKeyService(Database::connection());
    }

    public function get(Request $request, Response $response): Response
    {
        return $this->json($response, $this->service()->get($this->callerOrgId($request)));
    }

    public function generate(Request $request, Response $response): Response
    {
        return $this->json($response, $this->service()->generate($this->callerOrgId($request)));
    }

    public function revoke(Request $request, Response $response): Response
    {
        return $this->json($response, $this->service()->revoke($this->callerOrgId($request)));
    }
}

File: api/Enkl.Api/Controllers/OrganisationApiKeyController.cs
using Enkl.Api.Auth;
using Enkl.Api.Services;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace Enkl.Api.Controllers;

/// <summary>Same OrgAdmin gating as OrganisationSsoConfigController.</summary>
[ApiController]
[Authorize(Policy = "OrgAdmin")]
[Route("api/organisations/me/api-key")]
public class OrganisationApiKeyController : ControllerBase
{
    private readonly OrganisationApiKeyService _apiKey;

    public OrganisationApiKeyController(OrganisationApiKeyService apiKey)
    {
        _apiKey = apiKey;
    }

    [HttpGet]
    public async Task<IActionResult> Get()
    {
        return Ok(await _apiKey.GetAsync(User.OrgId()));
    }

    [HttpPost]
    public async Task<IActionResult> Generate()
    {
        return Ok(await _apiKey.GenerateAsync(User.OrgId()));
    }

    [HttpDelete]
    public async Task<IActionResult> Revoke()
    {
        return Ok(await _apiKey.RevokeAsync(User.OrgId()));
    }
}
